src/utils/__tests__/isStatelessComponent-test.js

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) 2015, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+/*global jest, describe, beforeEach, it, expect*/
+
+
+jest.autoMockOff();
+
+describe('isStatelessComponent', () => {
+  var isStatelessComponent;
+  var expression, statement, parse;
+
+  beforeEach(() => {
+    isStatelessComponent = require('../isStatelessComponent');
+    ({expression, statement, parse} = require('../../../tests/utils'));
+  });
+
+  describe('Stateless Function Components with JSX', () => {
+    it('accepts simple arrow function components', () => {
+      var def = parse(
+        'var Foo = () => <div />'
+      ).get('body', 0).get('declarations', [0]).get('init');
+      expect(isStatelessComponent(def)).toBe(true);
+    });
+
+    it('accepts simple function expressions components', () => {
+      var def = parse(
+        'let Foo = function() { return <div />; };'
+      ).get('body', 0).get('declarations', [0]).get('init');
+      expect(isStatelessComponent(def)).toBe(true);
+    });
+
+    it('accepts simple function declaration components', () => {
+      var def = parse('function Foo () { return <div /> }').get('body', 0);
+      expect(isStatelessComponent(def)).toBe(true);
+    });
+  });
+
+  describe('Stateless Function Components with React.createElement', () => {
+    it('accepts simple arrow function components', () => {
+      var def = parse(
+        'var Foo = () => React.creatElement("div", null);'
```

Comment:
Oh! Found my sticking point. Since I’m calling `recast.visit(node, {})` in the l`containsJSXElementOrReactCreateElementCall` function I’m losing all of recasts internal state